Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_Windows.cpp

```diff
@@ -149,6 +157,7 @@ namespace AzFramework
             AZ_Assert(result, "Failed to deregister raw input device: keyboard");
             AZ_UNUSED(result);
         }
+        s_instanceCount.Set(instanceCount);
```

Comment:
If we do need to reset the environment variable, we should probably do that at the end of the 'if' block above, then wrap this in an 'else' statement.